### PR TITLE
Velychkomaryna/feature/signup email confirmation integration

### PR DESCRIPTION
**Pull request by shuryhin-oleksandr**

### CODE REVIEW

File: src/app/(auth)/signup/components/SignUpEmailConfirmation.js [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
 import { SignUpContainer } from "./SignUpAccount";
+import { signupEmailConfirmation, USER_ID_KEY } from "@/app/api.mjs";
+import { Error } from "@/app/activities/[activityId]/edit/components/formikFields";
 
 export default function SignUpEmailConfirmation() {
+
+  const mutation = useMutation(signupEmailConfirmation);
+
+  const userId = localStorage.getItem(USER_ID_KEY);
+  const handleResendEmail = () => {
```
Comment (shuryhin-oleksandr) [inline src/app/(auth)/signup/components/SignUpEmailConfirmation.js:14]: Function does almost nothing, I would inline:
const mutation = useMutation(() => signupEmailConfirmation({userId}));

File: src/app/api.mjs [hunk 2]
```diff
@@ -15,6 +15,8 @@ const SCHOOLS_PATH = "/schools";
 const LOGIN_PATH = "/dj-rest-auth/login";
 const SIGNUP_ACCOUNT_PATH = "/dj-rest-auth/registration";
 const SIGNUP_DATAILS_PATH = "/signup-details";
+const SIGNUP_RESEND_EMAIL = "/resend-email";
+const CONFIRMATION_EMAIL = "/email-confirmation"
```
Comment (shuryhin-oleksandr) [inline src/app/api.mjs:19]: What is the purpose of this page?

File: src/app/confirmation/[confirmationKey]/page.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  return (
+    <Container sx={{ height: '100%', display: 'flex', justifyContent: 'center', alignItems: 'center', py: 10 }}>
+      {loading ? (
+        <CircularProgress />
+      ) : (
+        <Error>{mutation.isError && mutation.error.message}</Error>
+      )}
+    </Container>
+  );
+};
```
Comment (shuryhin-oleksandr) [inline src/app/confirmation/[confirmationKey]/page.js:41]: PEP-8 violation: no newline in the end of the file.

File: src/app/confirmation/[confirmationKey]/page.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+export default function EmailConfirmation({params}) {
+  const router = useRouter();
+  const [loading, setLoading] = useState(true);
+  const mutation = useMutation(emailConfirmation);
+  const { confirmationKey } = params;
+  const decodedKey = decodeURIComponent(confirmationKey);
+
+  useEffect(() => {
+    const confirmEmail = () => {
```
Comment (shuryhin-oleksandr) [inline src/app/confirmation/[confirmationKey]/page.js:18]: It is hard to read, as everything is in useEffect.
To make this simpler extract a function and just call it from a hook.
Defining functions inside methods or hooks is generally not the best idea.

File: src/app/api.mjs (hunk 2 above)
Comment (shuryhin-oleksandr) [inline src/app/api.mjs:19]: croos check naming with BE for consistency

File: src/app/confirmation/[confirmationKey]/page.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+import { useEffect, useState } from 'react';
+import { useRouter, useParams } from "next/navigation";
+import { useMutation } from "react-query";
+import { Error } from "@/app/activities/[activityId]/edit/components/formikFields";
+import { Container, CircularProgress } from '@mui/material';
+import { emailConfirmation } from "@/app/api.mjs";
+
+export default function EmailConfirmation({params}) {
+  const router = useRouter();
+  const [loading, setLoading] = useState(true);
```
Comment (shuryhin-oleksandr) [inline src/app/confirmation/[confirmationKey]/page.js:12]: loading is provided by react-query useMutation.

Comment (maryna-velychko) [reply src/app/confirmation/[confirmationKey]/page.js:12]: mutation.isLoading - is not the same visual effect. CircularProgress shows shorter than when I do it with state

Comment (maryna-velychko) [reply src/app/confirmation/[confirmationKey]/page.js:12]: in this case CircularProgress not visible to the user

File: src/app/confirmation/[confirmationKey]/page.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
+"use client";
+
+import { useEffect, useState } from 'react';
+import { useRouter, useParams } from "next/navigation";
+import { useMutation } from "react-query";
+import { Error } from "@/app/activities/[activityId]/edit/components/formikFields";
+import { Container, CircularProgress } from '@mui/material';
+import { emailConfirmation } from "@/app/api.mjs";
+
+export default function EmailConfirmation({params}) {
```
Comment (shuryhin-oleksandr) [inline src/app/confirmation/[confirmationKey]/page.js:10]: Why not just ?
`export default function EmailConfirmation({confirmationKey}) { `

File: src/app/api.mjs [hunk 7]
```diff
@@ -209,3 +211,15 @@ export async function signupDetails(userId, data) {
   const response = await client.put(url, data);
   return response.data;
 }
+
+export async function signupEmailConfirmation(data) {
```
Comment (shuryhin-oleksandr) [inline src/app/api.mjs:223]: Naming should be consistent across api functions, url paths, backend views

File: src/app/api.mjs [hunk 8]
```diff
@@ -209,3 +211,15 @@ export async function signupDetails(userId, data) {
...
+  const url = `${SIGNUP_RESEND_EMAIL}/`;
+  const response = await client.post(url, data);
+  return response.data;
+}
+
+export async function emailConfirmation(data) {
```
Comment (shuryhin-oleksandr) [inline src/app/api.mjs:229]: Same here

File: src/app/confirmation/[confirmationKey]/page.js (hunk 6 above)
Comment (maryna-velychko) [reply src/app/confirmation/[confirmationKey]/page.js:10]: It is not working. confirmationKey is undefined

File: src/app/confirmation/[confirmationKey]/page.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
+"use client";
+
+import { useEffect, useState } from 'react';
+import { useRouter } from "next/navigation";
+import { useMutation } from "react-query";
+import { Error } from "@/app/activities/[activityId]/edit/components/formikFields";
+import { Container, CircularProgress } from '@mui/material';
+import { verifyEmail } from "@/app/api.mjs";
+
+export default function EmailConfirmation({ params }) {
```
Comment (shuryhin-oleksandr) [inline src/app/confirmation/[confirmationKey]/page.js:10]: Naming not consistent with api function name.

File: src/app/confirmation/[confirmationKey]/page.js [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  const router = useRouter();
+  const [loading, setLoading] = useState(true);
+  const mutation = useMutation(verifyEmail);
+  const { confirmationKey } = params;
+  const decodedKey = decodeURIComponent(confirmationKey);
+
+  const confirmEmail = () => {
```
Comment (shuryhin-oleksandr) [inline src/app/confirmation/[confirmationKey]/page.js:17]: again, naming is not consistent with api. I'd name it handleVerifyEmail at least. As max, I would just write whole operation in useEffect.

**Pull request merged**